metricbeat/beater/metricbeat.go

```diff
@@ -20,72 +23,96 @@ Each event created has the following format:
 		"@timestamp": timestamp
 	}
 
-All documents are currently stored in one index called metricbeat. It is important to use an independent namespace
-for each MetricSet to prevent type conflicts. Also all values are stored under the same type "metricsets".
-
+All documents are stored in one index called metricbeat. It is important to use
+an independent namespace for each MetricSet to prevent type conflicts. Also all
+values are stored under the same type "metricsets".
 */
 package beater
 
 import (
-	"fmt"
+	"sync"
 
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/logp"
-	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/pkg/errors"
 )
 
+// Metricbeat implements the Beater interface for metricbeat.
 type Metricbeat struct {
-	done   chan struct{}
-	config *Config
+	done    chan struct{}    // Channel used to initiate shutdown.
+	config  *Config          // Metricbeat specific configuration data.
+	modules []*moduleWrapper // Active list of modules.
 }
 
 // New creates and returns a new Metricbeat instance.
 func New() *Metricbeat {
 	return &Metricbeat{}
 }
 
-func (mb *Metricbeat) Config(b *beat.Beat) error {
+// Config unpacks the Metricbeat specific configuration data.
+func (m *Metricbeat) Config(b *beat.Beat) error {
 	// List all registered modules and metricsets.
-	logp.Info("%s", helper.Registry.String())
+	logp.Info("%s", mb.Registry.String())
 
-	mb.config = &Config{}
-	err := b.RawConfig.Unpack(mb.config)
+	m.config = &Config{}
```

Comment:
Perhaps we should use in all beats `bt` for the beater to be consistent with beat-generator. Just as a note. No need to change.